scripts/zephyr_module.py

```diff
@@ -215,20 +220,52 @@ def main():
     cmake = ""
     sanitycheck = ""
 
+    dep_modules = []
+    sorted_modules = []
+    start_modules = []
+
     for project in projects:
         # Avoid including Zephyr base project as module.
         if project == os.environ.get('ZEPHYR_BASE'):
             continue
 
         meta = process_module(project)
         if meta:
-            kconfig += process_kconfig(project, meta)
-            cmake += process_cmake(project, meta)
-            sanitycheck += process_sanitycheck(project, meta)
+            section = meta.get('build', dict())
+            deps = section.get('depends', [])
+            if not deps:
+                start_modules.append( [project, meta] )
+            else:
+                dep_modules.append( [project, meta, deps] )
         elif project in extra_modules:
             sys.exit(f'{project}, given in ZEPHYR_EXTRA_MODULES, '
                      'is not a valid zephyr module')
 
+    # This will do a topology search to ensure the modules are ordered
+    # according to dependency settings.
+    while start_modules:
+        # We want to preserve order as close to that returned by west.
+        # Thus remove elements from beginning, instead of pop().
+        node = start_modules[0]
+        del start_modules[0]
+        sorted_modules.append(node)
+        node_name = PurePath(node[0]).name
+        for module in dep_modules:
+            if node_name in module[2]:
```

Comment:
Would you mind using a collections.namedtuple as a data structure just to make this more readable? I don't remember how this script works, and am feeling very lazy about chasing down what the the magic number indexes mean. It's okay if you don't have time; I'll make a better effort if that's the case :).

Reply:
good input, will clean up.
I was just too focused on getting things up and running, and still having a bit of C habits when writing python. Need to get out of dinosaur mode.

Reply:
Transition from dinosaur mode completed.